templates/common/baremetal/files/baremetal-mdns-publisher.yaml

```diff
@@ -34,11 +34,23 @@ contents:
         - "-c"
         - |
           #/bin/bash
-          while [ "$(hostname)" == "$DEFAULT_LOCAL_HOSTNAME" ]
+          function get_hostname()
+          {
+            if [[ -s /etc/mdns/hostname.txt ]]; then
+              echo $(</etc/mdns/hostname.txt)
+            else
+              # if hostname wasn't updated by NM script, read hostname
+              hostname
```

Comment:
What's the purpose of reading the cloned uts hostname?

Reply:
If verify-hostname init container start to run before NM dispatcher && If hostname in cloned uts namespace != localhost (meaning that verify-hostname start to run after hostname set by DHCP) - 
there is no reason to wait for NM script, right?